week3/homework/src/index.js

```diff
@@ -1,3 +1,133 @@
 'use strict';
 
-// Write the homework code in this file
+const express = require('express');
+const bodyParser = require('body-parser');
+const fs = require('fs');
+const uniqid = require('uniqid');
+
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(bodyParser.urlencoded({
+    extended: false
+}));
+app.use(bodyParser.json());
+
+const TODOS_FILENAME = {};
+const myTodoList = [];
+app.listen(port, (err) => {
+    if (err) {
+        console.log(err);
+    }
+    console.log(`Server running on port : ${port}`);
+});
+//_________________________________________________________________________
+
+const readTodosFile = (filename) => {
+    return new Promise((resolve, reject) => {
+        fs.readFile(filename, "utf8", (err, data) => {
+            if (err) {
+                if (err.code === "ENOENT") {
+                    console.log("Sorry, this file doesn't exist.");
+                } else if (err.code === "EACCES") {
+                    console.log("Sorry, but you don't have permission.");
+                } else {
+                    console.log(` unknown error occurred: ${err.message}`);
+                }
+                reject(err);
+            }
+            console.log(`Successfully read the file ${filename}`);
+            resolve(data);
+        });
+    });
+};
+const writeTodosFile = (filename, dataAsString) => {
+    return new Promise((resolve, reject) => {
+        fs.writeFile(filename, dataAsString, (err) => {
+            if (err) {
+                reject(err);
+            }
+            resolve(`The < ${filename}> file has been saved!`);
+        });
+    });
+};
+
+//add task
+const addTodoItem = (filename, todoList, todoDescription) => {
+    todoList.push({
+        "description": todoDescription,
+        "id": uniqid()
+    });
+    writeTodosFile(TODOS_FILENAME, JSON.stringify(myTodoList))
+        .then(() => console.log("Done adding todo."))
+        .catch((err) => {
+            console.log('ERROR:', err);
+        });
+};
+
+const removeTodoItem = (filename, todoList, todoId) => {
+    const index = todoList.map(x => {
+        return x.Id;
+    }).indexOf(todoId);
+    todoList.splice(index, 1);
+
+    todoList.filter(e => e.id !== todoId);
```

Comment:
You are not assigning the result of filtering to anything. What is the point of this?